_posts/2022-08-01-9.0.0-release.md

```diff
@@ -0,0 +1,164 @@
+to reduce the overhead of invoking compute functions and kernels, especially
+at the small data sizes desirable for efficient parallelization (typically
+L1- or L2-sized).  Specifically, the non-owning `ArraySpan` and `ExecSpan`
+structures have internally superseded the much heavier `ExecBatch`, which
+is still supported for compatibility at the API level
+(ARROW-16756, ARROW-16824, ARROW-16852).
+
+In a similar vein, the `ValueDescr` class was removed and `ScalarKernel`
+implementations now always receive at least one non-scalar input, removing
+the special case where a `ScalarKernel` needs to output a scalar rather than
+an array. The higher-level compute APIs still allow executing a scalar function
+over all-scalar inputs; but those scalars are internally broadcasted to
+1-element arrays so as to simplify kernel implementation (ARROW-16757).
+
+Timestamp comparison is now supported (ARROW-16425).
+
+A cumulative sum function is implemented over numeric inputs (ARROW-13530).
+
+Temporal rounding functions received additional options to control how
+rounding is done (ARROW-14821).
+
+Improper computation of the "mode" function on boolean input was fixed
+(ARROW-17096).
+
+Function registries can now be nested (ARROW-16677).
+
+### Dataset
+
+The `autogenerate_column_names` option for CSV reading is now handled correctly
+(ARROW-16436).
+
+Fix `InMemoryDataset::ReplaceSchema` to actually replace the schema
+(ARROW-16085).
+
+Fix `FilenamePartitioning` to properly support null values (ARROW-16302).
+
+### Filesystem
+
+A number of bugfixes and improvements were made to the Google Cloud Storage
+filesystem implementation (ARROW-14892).
+
+By default, the S3 filesystem implementation does not create or drop buckets
+anymore (ARROW-15906). This is a compatibility-breaking change intended
+to prevent user errors from having potentially catastrophic consequences.
+Options have been added to restore the previous behavior if necessary.
+
+### Parquet
+
+The default Parquet version is now 2.4 for writing, enabling use of
+more recent logical types by default (ARROW-12203).
+
+Non-nullable fields are now handled correctly by the Parquet reader
+(ARROW-16116).
+
+Reading encrypted files should now be thread-safe (ARROW-14114).
+
+Statistics equality now works correctly with minmax (ARROW-16487).
+
+The minimum Thrift version required for building is now 0.13 (ARROW-16721).
+
+The Thrift deserialization limits can now be configured to accomodate for
+data files with very large metadata (ARROW-16546).
+
+### Substrait
+
+The Substrait spec has been updated to 0.6.0 (ARROW-16816). In addition, a
+larger subset of the Substrait specification is now supported (ARROW-15587,
+ARROW-15590, ARROW-15901, ARROW-16657, ARROW-15591).
+
+## C# notes
+
+#### New Features
+
+* Added support for Add support for Time32Array and Time64Array ([ARROW-16660](https://github.com/apache/arrow/pull/13279))
+
+## Go notes
+
+## Java notes
+
```

Comment:
```suggestion
#### New Features
* Allow overriding column nullability in arrow-jdbc ([#13558](https://github.com/apache/arrow/pull/13558))
* Enable skip BOUNDS_CHECKING with setBytes and getBytes of ArrowBuf ([#13161](https://github.com/apache/arrow/pull/13161))
* Initialize JNI components on use instead of statically ([#13146](https://github.com/apache/arrow/pull/13146))
* Provide explicit JDBC column type mapping ([#13166](https://github.com/apache/arrow/pull/13166))
* Allow duplicated field names in Java C data interface ([#13247](https://github.com/apache/arrow/pull/13247))
* Improve and document StackTrace ([#12656](https://github.com/apache/arrow/pull/12656))
* Keep more context when marshaling errors through JNI ([#13246](https://github.com/apache/arrow/pull/13246))
* Make RoundingMode configurable to handle inconsistent scale in BigDecimals ([#13433](https://github.com/apache/arrow/pull/13433))
* Improve Java dev experience with IntelliJ ([#13017](https://github.com/apache/arrow/pull/13017))
* Implement ArrowArrayStream ([#13465](https://github.com/apache/arrow/pull/13465)))

#### Bug Fixes
* Fix variable-width vectors in integration JSON writer ([#13676](https://github.com/apache/arrow/pull/13676))
* Handle empty JDBC ResultSet ([#13049](https://github.com/apache/arrow/pull/13049))
* Fix hasNext() in ArrowVectorIterator ([#13107](https://github.com/apache/arrow/pull/13107))
* Fix ArrayConsumer when using ArrowVectorIterator ([#12692](https://github.com/apache/arrow/pull/12692))
* Update Gandiva Protobuf library to enable builds on Apple M1 ([#13121](https://github.com/apache/arrow/pull/13121))
* Patch dataset module testing failure with JSE11+ ([#13200](https://github.com/apache/arrow/pull/13200))
* Don't duplicate generated Protobuf classes between flight-core and flight-sql ([#13596](https://github.com/apache/arrow/pull/13596))
```